Sources/Datasets/datasets/structured/Iris.swift

```diff
@@ -0,0 +1,147 @@
+/// The Iris dataset.
+///
+/// This is perhaps the best known database to be found in the pattern recognition
+/// literature. Fisher's paper is a classic in the field and is referenced
+/// frequently to this day. (See Duda & Hart, for example.) The data set contains
+/// 3 classes of 50 instances each, where each class refers to a type of iris
+/// plant. One class is linearly separable from the other 2; the latter are NOT
+/// linearly separable from each other.
+/// 
+/// BibTeX citation:
+/// @misc{Dua:2019,
+///     author = "Dua, Dheeru and Graff, Casey",
+///     year = "2017",
+///     title = "{UCI} Machine Learning Repository",
+///     url = "http://archive.ics.uci.edu/ml",
+///     institution = "University of California, Irvine, School of Information and Computer Sciences"
+/// }
+
+import Foundation
+import S5TF
+import TensorFlow
+
+public struct Iris: S5TFDataset {
+    typealias DataLoader = IrisDataLoader
+    public static var train: IrisDataLoader {
+        get {
+            return IrisDataLoader()
+        }
+    }
+
+    public static var info = irisInfo
+    public static let numberOfTrainingExamples = 150
+    public static let numberOfFeatures = 4
```

Comment:
I don't now why, but these seem a bit odd, they just don't seem in place. We should perhaps encapsulate this in `S5TFDatasetInfo`.

Reply:
Hmm, I'm not sure about that because `Iris.info.numberOfFeatures` would be roundabout on one hand, but it makes sense if you think about it. However, it doesn't really fit with the other properties of `S5TFDatasetInfo`. I'm not sure what the correct approach is here.

Reply:
We could move properties such as number of features, number of training/test/validation to `S5TFDatasetInfo`, as we include more and more properties I think it woudl be a good place to store them. Another possibility is having another perhaps `S5TFDatasetFeatures` struct to store this kind of information.

Reply:
`S5TFDatasetFeatures` would be confusing with the "features" concept in ML.

Don't you think this adds too many abstractions?

Reply:
Perhaps, yeah. I still think we should at least put the properties it in the dataset info though.

Reply:
I think you already updated the requirements in https://github.com/s5tf-team/datasets/pull/7 no? I will update this accordingly.

Reply:
Please merge https://github.com/s5tf-team/datasets/pull/19.

Reply:
Would this be better in the `info` struct?